### PR TITLE
*: refactor DM faq and troubleshooting

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? 

Refactor DM FAQ and troubleshooting
- Update the position in `TOC.md`
- Modify the file path of `error-system.md` and `faq.md`
- Create a new directory `tools/data-migration/troubleshoot/`
- Reorganize the content according to new file names
- Add three new FAQ items
- Add three `aliases`

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs-cn/pull/1950/
### Which version does your change affect? 

dev, v3.1, v3.0, v2.1

### CODE REVIEW

File: dev/reference/tools/data-migration/faq.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+Currently, DM only supports decoding the standard version of MySQL or MariaDB binlog. It has not been tested for Alibaba Cloud RDS or other cloud databases. If you are confirmed that its binlog is in standard format, then it is supported.
+
+## Does the regular expression of the black and white list in the task configuration support `non-capturing (?!)`?
+
+Currently, DM does not support it and only supports the regular expressions of the Golang standard library. See regular expressions supported by Golang via [re2-syntax](https://github.com/google/re2/wiki/Syntax).
+
+## If a statement executed upstream contains multiple DDL operations, does DM support such replication?
+
+DM will attempt to split a single statement containing multiple DDL change operations into multiple statements containing only one DDL operation, but might not cover all cases. It is recommended to include only one DDL operation in a statement executed upstream, or verify it in the test environment. If it is not supported, you can file an [issue](https://github.com/pingcap/dm/issues) to the DM repository.
```
Comment (lilin90) [inline dev/reference/tools/data-migration/faq.md]: I think these three items are newly added to this repo. @WangXiangUSTC @yikeke Please take a look at the content here. Other contents in this PR are just reorganized based on existing files.

File: dev/reference/tools/data-migration/troubleshoot/dm.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+title: Data Migration Troubleshooting
+summary: Learn how to diagnose and resolve issues when you use Data Migration.
+category: reference
+aliases: ['/docs/dev/how-to/troubleshoot/data-migration/']
+---
+
+# Data Migration Troubleshooting
+
+This document summarizes some commonly encountered issues when you use Data Migration, and provides the solutions.
```
Comment (WangXiangUSTC) [inline dev/reference/tools/data-migration/troubleshoot/dm.md]: this document doesn't contain issues, only tell users how to check DM and then find the problem

File: dev/reference/tools/data-migration/troubleshoot/dm.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+If you encounter errors while running Data Migration, try the following solution:
+
+1. Execute the `query-status` command to view the task running status and related error output.
+
+2. Check the log content related to the error you encountered. The log files are on the DM-master and DM-worker deployment nodes. To ger key information about errors, see [common errors](/dev/reference/tools/data-migration/troubleshoot/error-system.md). Then check [error handling](/dev/reference/tools/data-migration/troubleshoot/error-handling.md) to find the corresponding solution.
```
Comment (WangXiangUSTC) [inline dev/reference/tools/data-migration/troubleshoot/dm.md]:
```suggestion
2. Check the log content related to the error you encountered. The log files are on the DM-master and DM-worker deployment nodes. To get key information about errors, see [common errors](/dev/reference/tools/data-migration/troubleshoot/error-system.md). Then check [error handling](/dev/reference/tools/data-migration/troubleshoot/error-handling.md) to find the corresponding solution.
```
do you mean get?

File: dev/reference/tools/data-migration/troubleshoot/dm.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+---
```
Comment (yikeke) [inline dev/reference/tools/data-migration/troubleshoot/dm.md]: 这个文件名有点奇怪，路径成了 `dev/reference/tools/data-migration/troubleshoot/dm.md`，dm 和前面的 data-migration 感觉有些重复了，要不改成 `dev/reference/tools/data-migration/troubleshoot/guide.md`？

Comment (yikeke) [reply dev/reference/tools/data-migration/troubleshoot/dm.md]: 哦，才发现中文pr已经合并了..

File: dev/reference/tools/data-migration/troubleshoot/dm.md (hunk 3 above)
Comment (lilin90) [reply dev/reference/tools/data-migration/troubleshoot/dm.md]: @WangXiangUSTC Exactly!

File: dev/reference/tools/data-migration/faq.md (hunk 1 above)
Comment (yikeke) [reply dev/reference/tools/data-migration/faq.md]: LGTM

**Comment by lilin90** [comment]
All comments addressed. Thanks for the review.

**Pull request merged**